Project Management Process.md

```diff
@@ -0,0 +1,168 @@
+- [Projects](#projects)
+- [Roles](#roles)
+  - [Product Owners](#product-owners)
+  - [Project Leads](#project-leads)
+  - [Project Managers](#project-managers)
+- [Structure](#structure)
+  - [Overview](#overview)
+  - [Goals](#goals)
+  - [Backlog](#backlog)
+  - [Milestones](#milestones)
+  - [Project Roadmap](#project-roadmap)
+  - [Organization Roadmap](#organization-roadmap)
+- [Flow](#flow)
+  - [Quarterly Planning](#quarterly-planning)
+  - [Software Development Pipeline](#software-development-pipeline)
+- [Implementation](#implementation)
+
+## Projects
+TODO
+
+## Roles
+TODO
+
+### Product Owners
+TODO
+
+### Project Leads
+TODO
+
+### Project Managers
+TODO
+
+## Structure
+
+### Overview
+The project information across the organization is tracked with the structure described in this chapter. At the highest level, we have an Organization Roadmap, which lists all projects and their milestones. Each project contains a set of milestone and milestones contain one or more goals. Goals are the basic unit of work and they describe everything from bugs and new features to general tasks. Goals are collected and tracked in a Backlog.
+
+```
+Organization        |   Project 1           Project 2
+                    |  
+Roadmap             |   Roadmap             Roadmap  
+                    |  
+  Milestones        |     Milestone 1         Milestone A
+    Milestone 1     |       Goal 1              Goal A
+    Milestone 2     |       Goal 2            Milestone B
+    Milestone 3     |       Goal 3              Goal B
+    Milestone A     |     Milestone 2           Goal C
+    Milestone B     |       Goal 4              Goal D
+    Milestone C     |     Milestone 3      
+                    |       Goal 5      
+  Projects          |  
+    Project 1       |   Backlog             Backlog
+    Project 2       |     Goal 1              Goal A
+    Project 3       |     Goal 2              Goal B
+                    |     Goal 3              Goal C
+                    |     Goal 4              Goal D
+                    |     Goal 5
+```
+
+### Goals
+At the heart of each project are the goals. They are the basic unit of work. Goals can be anything that ***adds value to a release***. This includes issues reported by users, feature requests, bugs, Pull-Requests, general tasks such as refactoring, documentation or research. Goals are collected, maintained and tracked in the backlog by the project lead.
+
+A goal description includes the following:
+- Name / Summary
+  - Describes what the goal is on high level
+  - Example 1: "Reduce the size of the go-ipfs executable"
+  - Example 2: "js-ipfs should have API Documentation"
+  - TODO: what is a good name for a goal?
+- Description
+  - What the problem is and how this goal solves it
+- Type
+  - Describe the type of this goal: is it a bug? a new feature? something else?
+- Status
+  - What is the current status of this goal. Is it to-be-reviewed? In progress? Blocked? Done?
```

Comment:
Okay, so this seems to be where the "stages of the kanban pipeline" get introduced. Please be explicit here that these are mutually exclusive states: an issue can only be one of "in progress", "blocked", "done". Also, express ALL the states, and define clear way to decide what state an issue belongs to. Basically:

---

- Status (kanban stage)
  - What is the current status of this goal? It can be one of:
    - `Icebox / Future Work` - the goal is identified and being fleshed out. It is not ready to be carried out at this moment.
    - `Backlog` - the goal is well defined and ready to be carried out, but not being actively worked on.
    - `In Progress` - the goal is actively being worked on, this moment.
    - `Blocked` - the goal cannot proceed as another goal must be Done first.
    - `Done` - the goal is completed. No further action is necessary.
  - These stages form the "Kanban Progress Pipeline"